contents/metropolis/code/python/metropolis.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+
+
+def f(x):
+    '''Function proportional to target distribution, a sum of Gaussians'''
+    # Gaussian heights, width parameters, and mean positions respectively:
+    a1, b1, x1 = 10,   4, -4
+    a2, b2, x2 =  3, 0.2, -1
+    a3, b3, x3 =  1,   2,  5
+    
+    return a1*np.exp( -b1*(x-x1)**2 ) + \
+           a2*np.exp( -b2*(x-x2)**2 ) + \
+           a3*np.exp( -b3*(x-x3)**2 )
+
+def g():
+    '''Random step vector'''
+    return np.random.uniform(-1,1)
+
+def iterate(x, f=f, g=g):
+    '''Perform one full iteration and return new position'''
+    
+    x_proposed = x + g()
+    A = min(1, f(x_proposed)/f(x) )
+    
+    u = np.random.uniform(0,1)
+    if u <= A:
+        x_new = x_proposed
+    else:
+        x_new = x
```

Comment:
Fair enough, but wouldn't `np.random.choice` be a better function for this?, something like this:
```suggestion
x_new = np.random.choice([x_proposed, x], p=[A, 1-A])

Reply:
Hmm, yeah that probably makes it easier to understand too. The whole `if u <= A` code block is just how the algorithm is written in pseudo code. But it would be far less confusing to say "choose proposed x with probability A", and then use your suggestion. This way we can do away with the variable `u` completely.